### ISSUE TITLE
invalid byte sequence in US-ASCII

### ISSUE THREAD

**Issue by verenion** [puphpet/puphpet#1792]
This is driving me insane.

Fresh copy of puphpet, hosted locally on virtualbox. ubuntu 32. I haven't touched ANYTHING. Running vagrant up works perfectly and the vm is accessible via http://192.168.56.101/ and http://awesome.dev/. However, if I then re-provision the vm using:

    vagrant reload --provision

I get an error after a few minutes with this log:

    ==> default: Notice: /Stage[main]/Pyenv/Vcsrepo[/usr/local/pyenv]/revision: revi
    sion changed '3cde540ca00208fbfd0a5e7708c3053bc8cca9f5' to 'v0.4.0-20140404'
    ==> default: Notice: /Stage[main]/Composer/Exec[download_composer]/returns: exec
    uted successfully
    ==> default: Notice: /Stage[main]/Puphpet_php/Puphpet::Php::Pecl[pecl_http]/Php:
    :Pecl::Module[pecl_http]/Exec[pecl-pecl_http]/returns: executed successfully
    ==> default: Info: Puphpet::Php::Pecl[pecl_http]: Scheduling refresh of Service[
    php5-fpm]
    ==> default: Error: /Stage[main]/Puphpet_php/Service[php5-fpm]: Could not evalua
    te: invalid byte sequence in US-ASCII
    ==> default: Notice: /Stage[main]/Puphpet_php/Service[php5-fpm]: Triggered 'refr
    esh' from 1 events
    ==> default: Notice: /Stage[main]/Puphpet_apache/File[/var/run/apache2/ssl_mutex
    ]/ensure: created
    ==> default: Info: /Stage[main]/Puphpet_apache/File[/var/run/apache2/ssl_mutex]:
     Scheduling refresh of Service[httpd]
    ==> default: Notice: /Stage[main]/Apache::Service/Service[httpd]: Triggered 'ref
    resh' from 1 events
    ==> default: Notice: Finished catalog run in 56.95 seconds
    The SSH command responded with a non-zero exit status. Vagrant
    assumes that this means the command failed. The output for this command
    should be in the log above. Please read the output to determine what
    went wrong.

TL;DR - I'm getting this error when I try and re-provision the vm after a successful boot. This happens regardless of whether I change anything or not. Just retried and literally didn't edit one file, same results. I don't even know where to look anymore, so any help would be appreciated.

**Comment by nickmartyn** [comment]
Hi *verenion*, i had the same problems with locale settings while re-provisioning box. I solved it by post-install script, named it set-locale.sh:

#!/bin/bash
echo "Setting locale to fix encoding"
sh -c "echo 'LC_ALL=en_US.UTF-8' >> /etc/environment"
echo "done setting locale"

 i put it to the folder: puphpet/files/exec-always. (And deleted empty file).

**Comment by verenion** [comment]
Thanks man, I appreciate the response! No luck I'm afraid. I literally just downloaded the latest puphpet using all the defaults on ubuntu 64. I vagrant up - perfect. When reprovisioning / rebooting, it just breaks. Same error everytime. I don't know where to look anymore!

**Comment by verenion** [comment]
So, after many, many, many attempts with different OS, reinstalling vagrant, virtualbox, messed around with various issues that I kept getting, including the one above, puphpet then failing to authenticate (see https://github.com/mitchellh/vagrant/issues/5186), then an issue with apache not starting. I've now just fixed the authentication issue and apache and it's all booting fine. After one reload, I get the original error again.

For those getting the error above, look at this page. You have to SSH into the server after it fails, edit the file to remove any strange characters and then reprovision it. That's when I got more issues. 

https://github.com/mayflower/puppet-php/issues/79

I've finally to give up. 

It's a big shame, because puphpet has massive potential - but I cannot commit to using software like this in a development environment. I've spent more time trying to fix it than actually using it.

**Comment by thiagotalma** [comment]
Try this adjustment: #1793

**Comment by verenion** [comment]
Thanks. Tried those changes. Didn't work.

Besides, this morning I'm getting a completely different issue when the vm boots.

    Warning: Authentication failure. Retrying..

**Comment by verenion** [comment]
Okay. I was running virtualbox 5 and vagrant 1.7.4.

I tried on another PC running 1.7.2 and virtualbox 4.3. It now seems to be working correctly. Is puphpet supposed to work on the latest versions? as everytime I try it on virtualbox 5, it just keeps creating errors.

**Comment by nickmartyn** [comment]
Me too, VirturalBox 5.0 & vagrant 1.7.4 + W8.1 x64 + puphpet ubuntu32 - nothing was booting. Switched back to VirtualBox 4.3

**Comment by jtreminio** [comment]
@verenion not a puphpet issue. Vagrant does not yet support Virtualbox 5.

**Comment by nickmartyn** [comment]
![vagrant vb support](https://cloud.githubusercontent.com/assets/8321669/8828770/7e607c00-309c-11e5-9797-f4c14ded2aec.PNG)

http://docs.vagrantup.com/v2/virtualbox
https://github.com/mitchellh/vagrant/pull/5603

**Comment by thiagotalma** [comment]
Please accept. Not a Puphpet problem.

https://github.com/mitchellh/vagrant/issues/5896

**Comment by thiagotalma** [comment]
@nickmartyn In fact, you can earn money if you want: https://twitter.com/mitchellh/status/618464817238667264 :wink:

**Comment by bezruki** [comment]
I believe the issues has to do with php5-fpm.conf file - the author's name - check this out:

https://github.com/mayflower/puppet-php/issues/79

Not sure what the fix it yet, but will post a comment if I can figure it out.

**Comment by stepansib** [comment]
I spent 2 days with this issue (and not only this) and already gone crazy
In fact i solved this issue by specifying default locale in config.yaml
```sh
locale:
    install: '1'
    settings:
        default_locale: en_US.UTF-8
        locales:
            - en_GB.UTF-8
            - en_US.UTF-8
```
After that machine ups and provisioning completes successfully.

**Comment by stuba** [comment]
Just wanted to point out that default_locale seems to have bee set before vagrant is created. I tried to fix provisioning problem by setting default_locale after vagrant box was created and it did not work. Good old vagrant destroy and vagrant up fixed problem.

Why is this not the default option? It seems that empty locale is not default by accident.

**Comment by robbyemmert** [comment]
Can you specify a UTF-8 locale, any UTF-8 locale as the default?  US-ASCII is pretty dangerous for this reason, even if it is the default for vagrant.

**Comment by kadiiskiFFW** [comment]
I have the same problem. And my ```default_locale``` is set. Any suggestions?
```
locale:
    install: '1'
    settings:
        default_locale: en_US.UTF-8
        locales:
            - en_GB.UTF-8
            - en_US.UTF-8
        timezone: Europe/Sofia
```
